pkg/util/containerd/containerd_util.go

```diff
@@ -0,0 +1,127 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build containerd
+
+package containerd
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/retry"
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/namespaces"
+)
+
+const (
+	// The check config is used if the containerd socket is detected.
+	// However we want to cover cases with custom config files.
+	containerdDefaultSocketPath = "/var/run/containerd/containerd.sock"
+)
+
+var (
+	globalContainerdUtil *ContainerdUtil
+	once                 sync.Once
+)
+
+// ContainerdItf is the interface implementing a subset of methods that leverage the Containerd api.
+type ContainerdItf interface {
```

Comment:
Very interesting article - I did not consider such an approach, I initially wanted to be consistent with the other checks and favoured the abstraction in the util rather than in the check. 
It might be an premature logic, I just liked the testability of my approach (I agree that I would probably have the same result if I moved the abstraction to the package) but also that using the Containerd API all the services are interfaces, so it seemed consistent to follow the same pattern